core/src/main/java/org/apache/calcite/sql/SqlCallBinding.java

```diff
@@ -398,10 +398,13 @@ public SqlCall permutedCall() {
    * @return signature exception
    */
   public CalciteException newValidationSignatureError() {
+    // String type = validator.getCatalogReader().getNamedType(call.getOperator()
+    // .getNameAsId()).getSqlTypeName().getName();
     return validator.newValidationError(call,
         RESOURCE.canNotApplyOp2Type(getOperator().getName(),
             call.getCallSignature(validator, scope),
-            getOperator().getAllowedSignatures()));
+            // getOperator().getAllowedSignatures()));
+            getOperator().getAllowedSignaturesUsingValidator(validator)));
```

Comment:
the idea is that this function will pass through validator which we'll use to check for all alias types for this `newValidationSignatureError`